x/protorev/keeper/routes.go

```diff
@@ -5,37 +5,31 @@ import (
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	gammtypes "github.com/osmosis-labs/osmosis/v14/x/gamm/types"
 	poolmanagertypes "github.com/osmosis-labs/osmosis/v14/x/poolmanager/types"
 	"github.com/osmosis-labs/osmosis/v14/x/protorev/types"
 )
 
-// BuildRoutes builds all of the possible arbitrage routes given the tokenIn, tokenOut and poolId that were used in the swap
-func (k Keeper) BuildRoutes(ctx sdk.Context, tokenIn, tokenOut string, poolId uint64, maxIterableRoutes *uint64) []poolmanagertypes.SwapAmountInRoutes {
+// BuildRoutes builds all of the possible arbitrage routes given the tokenIn, tokenOut and poolId that were used in the swap.
+func (k Keeper) BuildRoutes(ctx sdk.Context, tokenIn, tokenOut string, poolId uint64, maxPoolPoints *uint64) []poolmanagertypes.SwapAmountInRoutes {
 	routes := make([]poolmanagertypes.SwapAmountInRoutes, 0)
 
 	// Append hot routes if they exist
-	if tokenPairRoutes, err := k.BuildTokenPairRoutes(ctx, tokenIn, tokenOut, poolId, maxIterableRoutes); err == nil {
+	if tokenPairRoutes, err := k.BuildHotRoutes(ctx, tokenIn, tokenOut, poolId, maxPoolPoints); err == nil {
 		routes = append(routes, tokenPairRoutes...)
 	}
 
-	// Append an osmo route if one exists
-	if osmoRoute, err := k.BuildOsmoRoute(ctx, tokenIn, tokenOut, poolId, maxIterableRoutes); err == nil {
-		routes = append(routes, osmoRoute)
-	}
-
-	// Append an atom route if one exists
-	if atomRoute, err := k.BuildAtomRoute(ctx, tokenIn, tokenOut, poolId, maxIterableRoutes); err == nil {
-		routes = append(routes, atomRoute)
+	// Append highest liquidity routes if they exist
+	if highestLiquidityRoutes, err := k.BuildHighestLiquidityRoutes(ctx, tokenIn, tokenOut, poolId, maxPoolPoints); err == nil {
+		routes = append(routes, highestLiquidityRoutes...)
 	}
 
 	return routes
 }
 
-// BuildTokenPairRoutes builds all of the possible arbitrage routes from the hot routes given the tokenIn, tokenOut and poolId that were used in the swap
-func (k Keeper) BuildTokenPairRoutes(ctx sdk.Context, tokenIn, tokenOut string, poolId uint64, maxIterableRoutes *uint64) ([]poolmanagertypes.SwapAmountInRoutes, error) {
-	if *maxIterableRoutes <= 0 {
-		return []poolmanagertypes.SwapAmountInRoutes{}, fmt.Errorf("the number of routes that can be iterated through has been exceeded")
+// BuildHotRoutes builds all of the possible arbitrage routes using the hot routes method.
+func (k Keeper) BuildHotRoutes(ctx sdk.Context, tokenIn, tokenOut string, poolId uint64, maxPoolPoints *uint64) ([]poolmanagertypes.SwapAmountInRoutes, error) {
```

Comment:
Since `maxPoolPoints` is getting mutated in every function call, perhaps lets set rename it to `remainingPoolPoints`?